src/jimgw/core/single_event/transform_utils.py

```
@@ -0,0 +1,620 @@
+
+    Returns:
+        tuple[Float, Float]: Polar angle (theta) and azimuthal angle (phi).
+    """
+    phi = ra - gmst
+    theta = jnp.pi / 2 - dec
+    phi = (phi + 2 * jnp.pi) % (2 * jnp.pi)
+    return theta, phi
+
+
+def ra_dec_to_zenith_azimuth(
+    ra: Float, dec: Float, gmst: Float, rotation: Float[Array, "3 3"]
```

Ruff (0.15.15)

[error] Syntax error in forward annotation: Unexpected token at the end of an expression

(F722)